cookies/http-state/resources/cookie-setter.py

```diff
@@ -0,0 +1,57 @@
+from os import path;
+
+
+SETUP_FILE_TEMPLATE = "{}-test"
+EXPECTATION_FILE_TEMPLATE = "{}-expected"
+EXPECTATION_HTML_SCAFFOLD = "iframe-expectation-doc.html.py-str"
+DEBUGGING_HTML_SCAFFOLD = "debugging-single-test.html.py-str"
+DEFAULT_RESOURCE_DIR = path.join("cookies", "http-state", "resources")
+DEFAULT_TEST_DIR = "test-files"
+
+
+def dump_file(directory, filename):
+  return open(path.join(directory, filename), "r").read()
+
+
+class CookieTestResponse(object):
+  def __init__(self, file, root):
+    super(CookieTestResponse, self).__init__()
+    self.__test_file = SETUP_FILE_TEMPLATE.format(file)
+    self.__expectation_file = EXPECTATION_FILE_TEMPLATE.format(file)
+    self.__resources_dir = path.join(root, DEFAULT_RESOURCE_DIR)
+    self.__test_files_dir = path.join(self.__resources_dir, DEFAULT_TEST_DIR)
+
+  def cookie_setting_header(self):
+    return dump_file(self.__test_files_dir, self.__test_file)
+
+  def body_with_expectation(self):
+    html_frame = dump_file(self.__resources_dir, EXPECTATION_HTML_SCAFFOLD)
+    expected_data = dump_file(self.__test_files_dir, self.__expectation_file);
+    return html_frame.format(**{'data' : expected_data})
+
+
+def main(request, response):
+  if "debug" in request.GET:
+    response.writer.write_status(200)
+    response.writer.end_headers()
+    html_frame = dump_file(path.join(request.doc_root, DEFAULT_RESOURCE_DIR),
+                           DEBUGGING_HTML_SCAFFOLD)
+    test_file = html_frame % (request.GET['debug'])
+    response.writer.write_content(test_file)
+    response.writer.write()
+    return;
+
+  if not "file" in request.GET:
+    response.writer.write_status(404)
+    response.writer.end_headers()
+    response.writer.write_content("The 'file' parameter is missing!")
+    response.writer.write()
+    return;
+
+  cookie_response = CookieTestResponse(request.GET['file'], request.doc_root)
+
+  response.writer.write_status(200)
+  response.writer.write(cookie_response.cookie_setting_header())
+  response.writer.end_headers()
+  response.writer.write_content(cookie_response.body_with_expectation())
+  response.writer.write()
```

Comment:
@FHorschig I do not appear to be able to leave comments in https://chromium-review.googlesource.com/c/chromium/src/+/807988 so I have to provide feedback here instead.



All of the uses of write() in this file make me suspicious. When we run these tests in Firefox we get a lot of:

[task 2018-02-16T18:29:59.280Z] 18:29:59  WARNING - Traceback (most recent call last):
[task 2018-02-16T18:29:59.281Z] 18:29:59  WARNING -   File "/builds/worker/workspace/build/venv/lib/python2.7/site-packages/wptserve/handlers.py", line 254, in __call__
[task 2018-02-16T18:29:59.282Z] 18:29:59  WARNING -     rv = self.func(request, response)
[task 2018-02-16T18:29:59.283Z] 18:29:59  WARNING -   File "/builds/worker/workspace/build/tests/web-platform/tests/cookies/http-state/resources/cookie-setter.py", line 57, in main
[task 2018-02-16T18:29:59.283Z] 18:29:59  WARNING -     response.writer.write()
[task 2018-02-16T18:29:59.284Z] 18:29:59  WARNING - TypeError: write() takes exactly 2 arguments (1 given)

Did you mean to use [response.write()](http://wptserve.readthedocs.io/en/latest/response.html#wptserve.response.Response.write) instead?

Reply:
Thanks for your comment, I added it to the review on your behalf. I will have a look into that later this week.

Reply:
I dropped the failing calls entirely. The `write_content(...)` call is all we need. 

(In fact, `response.write()` ends by calling `response.writer.write_content(...)`).

Reply:
Hi @jdm , looks like @FHorschig has addressed your comments (see https://chromium-review.googlesource.com/c/chromium/src/+/807988/12..15/third_party/WebKit/LayoutTests/external/wpt/cookies/http-state/resources/cookie-setter.py ). If that looks good to you, could you r+ to unblock the merge? Thanks!